### PR TITLE
enable optuna multi-objectives feature

### PR DESCRIPTION
**Pull request by sywangyi**
# What does this PR do?

Fixes # ((https://github.com/huggingface/transformers/issues/25657))

## Who can review?
@sgugger  @ArthurZucker 

 -->

### CODE REVIEW

File: src/transformers/integrations/integration_utils.py [hunk 1]
```diff
@@ -205,10 +205,17 @@ def _objective(trial, checkpoint_dir=None):
 
         timeout = kwargs.pop("timeout", None)
         n_jobs = kwargs.pop("n_jobs", 1)
-        study = optuna.create_study(direction=direction, **kwargs)
+        directions = kwargs.pop("directions", None)
+        if directions is not None:
+            direction = None
```
Comment (ArthurZucker) [inline src/transformers/integrations/integration_utils.py:210]:
```suggestion
        direction = None if directions is not None else direction
```

File: src/transformers/trainer.py [hunk 2]
```diff
@@ -2515,7 +2516,7 @@ def hyperparameter_search(
         backend: Optional[Union["str", HPSearchBackend]] = None,
         hp_name: Optional[Callable[["optuna.Trial"], str]] = None,
         **kwargs,
-    ) -> BestRun:
+    ) -> Union[BestRun, List[BestRun]]:
         """
         Launch an hyperparameter search using `optuna` or `Ray Tune` or `SigOpt`. The optimized quantity is determined
         by `compute_objective`, which defaults to a function returning the evaluation loss when no metric is provided,
```
Comment (ArthurZucker) [inline src/transformers/trainer.py:2522]: documentation should be updated

**Review by ArthurZucker — COMMENTED** [review]
Thanks for working on this! Left a few nits

File: src/transformers/trainer.py (hunk 2 above)
Comment (sywangyi) [reply src/transformers/trainer.py:2522]: update the hpo_train.md

Comment (ArthurZucker) [reply src/transformers/trainer.py:2532]: What I mean here is that the `Return` type in the docstring as well as the `directions` argument should also be updated/added to mention this! 😉

**Review by ArthurZucker — COMMENTED** [review]
Okay! Just left a small nit, since `create_study` take direction and directions (which is not really amazing in terms of API) let's stick with this. Just need to add `direction` to the docstring and good to go !

**Comment by amyeroberts** [comment]
cc @muellerzr @pacman100 for input on the changes to trainer

**Review by muellerzr — APPROVED** [review]
Thanks! Trainer changes look good to me

**Review by ArthurZucker — APPROVED** [review]
Thanks for iterating !

File: src/transformers/trainer.py [hunk 3]
```diff
@@ -2553,6 +2554,9 @@ def hyperparameter_search(
             direction (`str`, *optional*, defaults to `"minimize"`):
                 Whether to optimize greater or lower objects. Can be `"minimize"` or `"maximize"`, you should pick
                 `"minimize"` when optimizing the validation loss, `"maximize"` when optimizing one or several metrics.
+            directions (`List[str]`, *optional*, defaults to None):
+                Whether to optimize multi objectives. Can be List of `"minimize"` and `"maximize"`, you should pick
+                `"minimize"` when optimizing the validation loss, `"maximize"` when optimizing one or several metrics.
```
Comment (amyeroberts) [inline src/transformers/trainer.py:2559]: Why do we add directions here? Couldn't we allow `direction` to be a string or list of strings?

Comment (sywangyi) [reply src/transformers/trainer.py:2559]: aligned with https://github.com/optuna/optuna/blob/7b3824aaab38843fdfbfc67ea93d6da446dc1548/optuna/study/study.py#L1125

Comment (sywangyi) [reply src/transformers/trainer.py]: done

Comment (sywangyi) [reply src/transformers/trainer.py]: change direction to be a string or list of strings?

File: src/transformers/integrations/integration_utils.py [hunk 4]
```diff
@@ -205,7 +205,7 @@ def _objective(trial, checkpoint_dir=None):
 
         timeout = kwargs.pop("timeout", None)
         n_jobs = kwargs.pop("n_jobs", 1)
-        directions = kwargs.pop("directions", None)
+        directions = direction if type(direction) is list else None
```
Comment (amyeroberts) [inline src/transformers/integrations/integration_utils.py:208]:
```suggestion
        directions = direction if isinstance(direction, list) else None
```

**Review by amyeroberts — APPROVED** [review]
Thanks for adding this and iterating!

**Pull request merged**